plugins/defaultplugins/ifplugin/interface_config.go

```diff
@@ -673,37 +667,34 @@ Modify rx-mode on specified VPP interface
 */
 func (plugin *InterfaceConfigurator) modifyRxModeForInterfaces(oldIntf *intf.Interfaces_Interface, newIntf *intf.Interfaces_Interface,
 	ifIdx uint32) error {
-	oldRxSettings := oldIntf.RxModeSettings
-	newRxSettings := newIntf.RxModeSettings
-	if oldRxSettings != newRxSettings {
-		var oldRxMode intf.RxModeType
-		if oldRxSettings != nil {
-			oldRxMode = oldRxSettings.RxMode
-		}
-		if newRxSettings != nil {
+	oldRx := oldIntf.RxModeSettings
+	newRx := newIntf.RxModeSettings
+
+	if oldRx == nil && newRx != nil || oldRx != nil && newRx == nil || *oldRx != *newRx {
+		if newRx != nil {
 			switch newIntf.Type {
 			case intf.InterfaceType_ETHERNET_CSMACD:
-				if newRxSettings.RxMode == intf.RxModeType_POLLING {
-					return plugin.modifyRxMode(ifIdx, newIntf, oldRxMode, *newRxSettings)
+				if newRx.RxMode == intf.RxModeType_POLLING {
+					return plugin.modifyRxMode(ifIdx, newIntf, oldRx.RxMode, *newRx)
 				}
-				plugin.log.WithFields(logging.Fields{"rx-mode": newRxSettings.RxMode}).
+				plugin.log.WithFields(logging.Fields{"rx-mode": newRx.RxMode}).
 					Warn("Attempt to set unsupported rx-mode on Ethernet interface.")
 			default:
-				return plugin.modifyRxMode(ifIdx, newIntf, oldRxMode, *newRxSettings)
+				return plugin.modifyRxMode(ifIdx, newIntf, oldRx.RxMode, *newRx)
```

Comment:
Accessing fields from `oldRx` is not safe since it can be zero. This was previously guarded by nil check: https://github.com/ligato/vpp-agent/pull/575/files#diff-17be73b9c784e4fd3438f6d99cc62433L680

I committed failing test case for this: https://github.com/ligato/vpp-agent/pull/575/commits/6c1a6ea53135c716e71b5235303e1a6e2b1a0409